integration-tests/README.md

```diff
@@ -0,0 +1,121 @@
+to build enclosed Quarkus, Micronaut and Helidon projects.
+
+## RuntimesSmokeTest
+
+The goal is to build and start applications with some real source code that actually
+exercises some rudimentary business logic of selected libraries.
+
+### Collect results:
+
+There are several log files archived in `testsuite/target/archived-logs/` after runtimes test execution. e.g.:
+
+```
+org.graalvm.tests.integration.RuntimesSmokeTest/quarkusFullMicroProfile/report.md
+org.graalvm.tests.integration.RuntimesSmokeTest/quarkusFullMicroProfile/build-and-run.log
+org.graalvm.tests.integration.RuntimesSmokeTest/quarkusFullMicroProfile/measurements.csv
+```
+
+`report.md` is human readable description of what commands were executed and how much time and memory was spent to
+get the expected output form the runtime's web server. `measurements.csv` contains the same data in a machine friendly form.
+Last but not least, `build-and-run.log` journals the whole history of all commands and their output.
+
+There is also an aggregated `testsuite/target/archived-logs/aggregated-report.md` describing all the testsuite did.
+
+## AppReproducersTest
+
+This part of the test suite runs smaller apps that are not expected to offer a web server, so just their stdout/stderr
+is checked. No measurements are made as to the used memory at the time of writing, but it could be easily changed.
+e.g. a current example:
+
+```
+org.graalvm.tests.integration.AppReproducersTest/randomNumbersReinit/report.md
+org.graalvm.tests.integration.AppReproducersTest/randomNumbersReinit/build-and-run.log
+``` 
+
+## Logs and Whitelist
+
+Logs are checked for error and warning messages. Expected error messages can be whitelisted in [WhitelistLogLines.java](./testsuite/src/it/java/org/graalvm/tests/integration/utils/WhitelistLogLines.java).
+
+## Thresholds
+
+The test suite works with ```threshold.properties```, e.g. `./apps/quarkus-full-microprofile/threshold.properties`: 
+
+```
+linux.time.to.first.ok.request.threshold.ms=50
+linux.RSS.threshold.kB=120000
+windows.time.to.first.ok.request.threshold.ms=80
+windows.RSS.threshold.kB=120000
+```
+
+**THIS IS NOT A PERFORMANCE TEST** The thresholds are in place only as a sanity check to make sure
+an update to Native image did not make the application runtime to run way over the expected benevolent values.
+
+The measured values are simply compared to be less or equal to the set threshold. One can overwrite the `threshold.properties`
+by using env variables or system properties (in this order). All letter are capitalized and dot is replaced with underscore, e.g.
+
+```
+APPS_QUARKUS_FULL_MICROPROFILE_LINUX_TIME_TO_FIRST_OK_REQUEST_THRESHOLD_MS=35 mvn clean verify -Ptestsuite 
+```
+
+### Example failures
+
+With a rather harsh threshold of 5ms:
+
+```
+org.opentest4j.AssertionFailedError: 
+Application QUARKUS_FULL_MICROPROFILE took 27 ms to get the first OK request, 
+which is over 5 ms threshold. ==> expected: <true> but was: <false>
```

Comment:
The `expected: <true> but was: <false>` is slightly confusing, I guess it refers to `27 <= 5`?
Unfortunately I can only think about adding the actual check just before `==>`,  to improve this.

Reply:
@zakkak junit.jupiter.api doesn't give me GreaterThan / LessThan IMHO... So it boils down to this:

```
assertTrue(timeToFirstOKRequest <= timeToFirstOKRequestThresholdMs,
               "Application " + app + " took " + timeToFirstOKRequest
                + " ms to get the first OK request, which is over " +
                timeToFirstOKRequestThresholdMs + " ms threshold.");
```

Reply:
Thought so. What I am suggesting is:
```java
assertTrue(timeToFirstOKRequest <= timeToFirstOKRequestThresholdMs,
               "Application " + app + " took " + timeToFirstOKRequest
                + " ms to get the first OK request, which is over " +
                timeToFirstOKRequestThresholdMs + " ms threshold. (" + app + " <= " + timeToFirstOKRequest + ")");
```
But that's probably more confusing.

There is a better approach discussed [here](https://stackoverflow.com/a/18766311/1119431) but I am probably nit-picking.